### DISCUSSION TITLE
Panel 3 Q/A

### DISCUSSION THREAD

**Discussion by cmbarton** [cssi-pi-community/2022-meeting#9] [Panelist Q/A for all discussions]
### Post a question for panelists in a comment responding to this discussion thread.

**Comment by stefanhenneking** [comment]
To promote sustainability, panelist M. Zenter (SDSC) suggested rewarding "non-reinvention." 
Can you elaborate further on this point, and maybe share ideas how to implement such a reward system?

**Reply by munkm** [reply]
I'm very interested in this too. What ideas does everybody have about this? I can imagine in proposals or plans that one could describe the following: 
* Specifically naming what software you plan to build off of with your tools
* How you plan to contribute to the tools you're using (with time, bugfixes, issues, or financial resources) in addition to what you're developing
* How you plan to share your own tool with the community so others can build off of your tool
* How your project will build community investment/trust/interaction (through defined governance, user workshops, developer workshops, etc).

**Comment by cmbarton** [comment]
The issue from my experience is that funding opportunities from NSF almost always assume that something **new** is being proposed. There is no explicit opportunity to apply for funding based on continuing and improving on what has been done in a prior award. Using results of prior NSF research is sometimes mentioned in RFPs but it rarely counts for much. So rather than a guarantee of support for sustainability it is important for there to be explicit **opportunity** to make the case for continuing infrastructure rather than assuming implicitly or explicitly that all funding is for new 'research'.

**Reply by munkm** [reply]
> almost always assume that something new is being proposed.

Yes! So much in software development can't be framed under a scientific question, or is challenging to frame this way. So we're going to have trouble with sustainability unless it can be framed well with this context.

**Reply by pibion** [reply]
Also, I get that funding is limited.  But I would love to see these maintenance projects compete on an even playing field.  If they're supporting a huge range of science and a large user base ... I'm okay with them edging out my new project!

**Reply by pibion** [reply]
Current projects that have been mentioned as supporting existing projects: TIP, POSE, CSSI path to sustainability (is that POSE?)

**Reply by pibion** [reply]
The I-Corps program (https://beta.nsf.gov/funding/initiatives/i-corps)

**Comment by munkm** [comment]
I'm curious what models the panelists are familiar with for project/software maintenance (which is so key for long-term software sustainability) beyond the initial 3-5 years of an award. Some of the panelists discussed longer-term projects that were able to last longer through other modes, and I'd love to hear more about that. So much research software is niche and financial support is possible primarily through grants and finding a way to sustain financial support for niche tools outside of grants would be helpful. In my experience, this maintenance ends up being done in spare time outside of work, which in itself is an issue because only people who have the time and resources to do so can do this longer-term.

**Comment by erjank** [comment]
Presently it seems there aren't reward structures that encourage _playing well with others_, like Michael said, the _who_ of sustainability: Are there funding examples that we're not aware of, or are there funding mechanisms we might push for that would support cross-collaboration between communities like NanoHub, MolSSI, etc.

**Comment by pibion** [comment]
On education in university courses - I use a finite-element approach in my intro physics class, and require advanced students to do a programming project while using github.

This gives me a chance to train students on skills they'll need in research and practice good practices but doesn't require invention of new courses.

**Reply by pibion** [reply]
I'd also love to see a learning assistant approach to more of our dedicated software training (https://www.learningassistantalliance.org/).

**Comment by rapier1** [comment]
It seems to me that there are certain inefficiencies in the process. We often get two years of funding which is usually the minimum amount of time to simply move through the process of design, implementation, testing, documentation, and packaging. At that point the grant runs out and maintaining the  code becomes a volunteer, at best, effort. The idea that we can develop a broad and deep user base that will, in turn, help maintain the code is often a wildly optimistic assumption. As such it's not uncommon for the initial investment by the NSF to have a limited impact.

**Reply by rapier1** [reply]
Also, just as a matter of awareness - no small number of people developing scientific applications are soft money funded staff. That can add another layer of complexity into the process.

**Reply by munkm** [reply]
> At that point the grant runs out and maintaining the code becomes a volunteer, at best, effort. 

Yes. And only those people with the resources to volunteer their time end up contributing longer-term. This leaves out anybody with limited financial resources, people with family commitments, or people with dependents (among many others).

**Reply by rapier1** [reply]
https://xkcd.com/2347/

**Reply by munkm** [reply]
> The idea that we can develop a broad and deep user base that will, in turn, help maintain the code is often a wildly optimistic assumption. 

Something that I am struggling with is also expecting the community to take up maintenance. Is that fair and equitable? How are those people paid? Do we leave out people when we do this? How do we ensure community governance and culture is maintained without centralized support?

**Reply by rapier1** [reply]
You'd still need someone who has the authority to handle PRs and other management issues. Which would be another volunteer.

**Comment by mbrazil1** [comment]
The dissemination of information and raising awareness about research projects (highlighting successes, publications and impact stories) is something that is often overlooked and could benefit from dedicated attention. Demonstrating the success of projects may help with sustainability. I suggest looking to your marketing departments at your institutions, students, etc. to help with some of these efforts (or potentially write a small percentage of effort into proposals).
